core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/MergeLimitWithSort.java

```diff
@@ -33,6 +34,7 @@
 
     private static final Pattern<LimitNode> PATTERN = limit()
             .matching(limit -> !limit.isWithTies())
+            .with(isOrderSensitive().equalTo(false))
```

Comment:
This should be allowed as long as the ordering scheme is compatible (which should always be the case unless there's a bug)